src/main/java/org/jabref/logic/openoffice/style/StyleLoader.java

```diff
@@ -129,17 +136,78 @@ public boolean removeStyle(OOBibStyle style) {
         return false;
     }
 
-    public OOBibStyle getUsedStyle() {
-        String filename = openOfficePreferences.getCurrentStyle();
+    public StyleIdentifier getUsedStyle() {
+        StyleSelectDialogViewModel.StyleType lastUsedStyleType = openOfficePreferences.getCurrentStyleType();
+
+        if (lastUsedStyleType == StyleSelectDialogViewModel.StyleType.CSL) {
+            String cslStyleName = openOfficePreferences.getCurrentCslStyleName();
+            if (cslStyleName != null && !cslStyleName.isEmpty()) {
+                // Set the CSL style using CSLCitationOOAdapter
+                CSLCitationOOAdapter.setSelectedStyleName(cslStyleName);
+                LOGGER.info("CSL style '{}' has been set", cslStyleName);
+
+                return new StyleIdentifier(StyleSelectDialogViewModel.StyleType.CSL, cslStyleName);
+            }
+        }
+
+        // This is the existing logic for JStyle
+        String filename = openOfficePreferences.getCurrentJStyle();
         if (filename != null) {
             for (OOBibStyle style : getStyles()) {
                 if (filename.equals(style.getPath())) {
-                    return style;
+                    return new StyleIdentifier(StyleSelectDialogViewModel.StyleType.JSTYLE, style);
                 }
             }
         }
-        // Pick the first internal
-        openOfficePreferences.setCurrentStyle(internalStyles.getFirst().getPath());
-        return internalStyles.getFirst();
+
+        // If no valid style is found, pick the first internal style as default
+        OOBibStyle defaultStyle = internalStyles.getFirst();
+        openOfficePreferences.setCurrentJStyle(defaultStyle.getPath());
+        openOfficePreferences.setCurrentStyleType(StyleSelectDialogViewModel.StyleType.JSTYLE);
+        return new StyleIdentifier(StyleSelectDialogViewModel.StyleType.JSTYLE, defaultStyle);
+    }
+
+    /**
+     * @param style This can be either a String (for CSL) or OOBibStyle (for JStyle)
+     */
+    public class StyleIdentifier {
```

Comment:
You now run into the architecture issues I already talked about in the beginning.

Make this an interface.
Let both CSLStyle and OOStyle implement the interface
In the StyleSelectionViewMOdel you then store he concrete style object 
https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof-operator.html 

As an example you can take a look at the PreviewLayout interface and it's subclasses, e.g. CitationStylePreviewLayout   
The latter one also has a path method 


I think this makes it a bit more cleaner and you don't have to do a cast to object